Build warning in `test/fuzz/addrdb.cpp`: unused variable

Seen both locally and in the Travis fuzz build:
```cpp
test/fuzz/addrdb.cpp:20:21: warning: unused variable 'ban_entry' [-Wunused-variable]
    const CBanEntry ban_entry = [&] {
                    ^
1 warning generated.
```

--- a/src/test/fuzz/addrdb.cpp
+++ b/src/test/fuzz/addrdb.cpp
@@ -17,6 +17,7 @@ void test_one_input(const std::vector<uint8_t>& buffer)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
 
+    // The point of this code is to exercise all CBanEntry constructors.
     const CBanEntry ban_entry = [&] {
         switch (fuzzed_data_provider.ConsumeIntegralInRange<int>(0, 2)) {
         case 0:
@@ -32,4 +33,5 @@ void test_one_input(const std::vector<uint8_t>& buffer)
         }
         return CBanEntry{};
     }();
+    (void)ban_entry; // currently unused
 }

